### ISSUE TITLE
Uniform prior not reporting the expected value

### ISSUE THREAD

**Issue by facero** [gammapy/gammapy#5447]
**Gammapy version**
gammapy-dev

**Bug description**
The Uniform prior is not consistent with the Gaussian prior

**Expected behavior**
In the Gaussian prior, the reported prior term is zero at the prefered value and increase when deviating from it (as expected).
However in the Uniform prior, the reported value is 1 (later multiplied by the weight) in the min, max range and zero outside.
This is conterintuitive and inconsistent with the Gaussian prior.

**To Reproduce**
See docs :
https://docs.gammapy.org/1.2/tutorials/api/priors.html?highlight=priors

**Possible fix** 
I think this is just a typo in the line definition :  
https://github.com/gammapy/gammapy/blob/71089b6eca53f45bbe05bdb751a41da399c1f2ee/gammapy/modeling/models/prior.py#L181

Basically if one inverts one and zeros in the `evaluate` it will work as expected : 

```

"""Evaluate the uniform prior."""
        if min < value < max:
            return 1.0 # should be 0
        else:
            return 0.0 # should be 1
```

I'll open a PR if you all agree.
